packages/docs-app/src/examples/core-examples/common/sizeSelect.tsx

```diff
@@ -27,25 +27,21 @@ export interface SizeSelectProps {
     onChange: (size: Size) => void;
 }
 
-export const SizeSelect: React.FC<SizeSelectProps> = ({ label, size, optionLabels, onChange }) => {
-    const handleChange = React.useCallback((value: string) => onChange(value as Size), [onChange]);
-
-    return (
-        <FormGroup label={label}>
-            <SegmentedControl
-                fill={true}
-                small={true}
-                options={[
-                    { label: optionLabels[0], value: "small" },
-                    { label: optionLabels[1], value: "regular" },
-                    { label: optionLabels[2], value: "large" },
-                ]}
-                onValueChange={handleChange}
-                value={size}
-            />
-        </FormGroup>
-    );
-};
+export const SizeSelect: React.FC<SizeSelectProps> = ({ label, size, optionLabels, onChange }) => (
+    <FormGroup label={label}>
+        <SegmentedControl<Size>
```

Comment:
Out of curiosity, in these types of cases if you omit `<Size>` is the generic type parameter properly inferred from the `options` array? I'm guessing not because there's no `as const`.

Reply:
>  I'm guessing not because there's no as const.

Correct, it can't be inferred from the `options` in that case. But it _is_ inferred from the type of the `value` prop as well as the arg of the `onChange` prop